### ISSUE TITLE
Cache capacity is twice of the expected size

### ISSUE THREAD

**Issue by Chuckame** [facebook/rocksdb#10707]
### Expected behavior
I want the max shared cache capacity to 1gb including 150mb mem tables. 
To do that, I use this config (jvm) that is coming from kafka streams, that is the same as lot of other libs using rocksdb, even not in Java :
```kotlin

class BoundedMemoryRocksDBConfigSetter : RocksDBConfigSetter {
    override fun setConfig(storeName: String, options: Options, configs: Map<String, Any>) {
        options.updateTableFormatConfig {
            setBlockCache(cache)
            setCacheIndexAndFilterBlocks(true)
            setCacheIndexAndFilterBlocksWithHighPriority(true)
            setPinTopLevelIndexAndFilter(true)
            setOptimizeFiltersForMemory(true)
        }

        options.setWriteBufferManager(writeBufferManager)
        options.setMaxWriteBufferNumber(3) // kafka streams default
        options.setCompactionStyle(CompactionStyle.UNIVERSAL) // kafka streams default
        options.setCompressionType(CompressionType.LZ4_COMPRESSION)
        options.setStatsPersistPeriodSec(0) // disable dumping stats, unable to have them locally btw
    }

    private fun Options.updateTableFormatConfig(block: BlockBasedTableConfig.() -> Unit) {
        setTableFormatConfig((this.tableFormatConfig() as BlockBasedTableConfig).apply(block))
    }

    override fun close(storeName: String, options: Options) {
        // nothing to clean
    }

    companion object {
        val cache = LRUCache(
            /* capacity = */ 1 * 1024 * 1024 * 1024,
            /* numShardBits = */ -1,
            /* strictCapacityLimit = */ false,
            /* highPriPoolRatio = */ 0.1,
        )
        val writeBufferManager = WriteBufferManager(
            /* bufferSizeBytes = */ 150 * 1024 * 1024,
            /* cache = */ cache,
            /* allowStall = */ true,
        )
    }
}
```

It's deployed inside a container in kubernetes.

But after that, the rocksdb cache capacity indicates the twice (2gb). And the cache usage is near to 2gb.

### Actual behavior
Get capacity returns exactly twice of the configured size, e.g. Returns 2gb when 1gb configured, 600mb when 300mb configured, 400 when 200 configured.... 

### Steps to reproduce the behavior
Just use this config and it should be the same ?

**Comment by riversand963** [comment]
Can you set `strictCapacityLimit` true?

**Comment by Chuckame** [comment]
Yes, I've put strict limit to true and false, stall writes to true and false. Tried to allow a capacity of 1gb on 2.5gb available, and 2gb on 2.5gb available, it's always the same : twice of the configured capacity.

**Comment by dkarpas** [comment]
Shared this issue with the Speedb hive (on [Discourse](https://speedb.discourse.group/t/cache-capacity-is-twice-of-the-expected-size-10707/44/3?u=danc)) and this is the buzz:

"from what i can tell, kafka streams creates a new block cache for each RocksDB state store. so could it be that you’re not passing the same cache object as specified here - [Kafka Streams Memory Management | Confluent Documentation](https://docs.confluent.io/platform/current/streams/developer-guide/memory-mgmt.html?&_ga=2.252237065.1297392088.1666463104-1065181344.1666463104#rocksdb)"

If you have follow up questions or want more info and help, join our hive on [Discord](https://discord.gg/gRguBYXhGH) and [Discourse](https://speedb.discourse.group/invites/KnAHZ3kjCg) 
To send the LOG file (preferably with statistics enabled) for review, send it to me @ dan@speedb.io

**Comment by Chuckame** [comment]
@dkarpas My code is already coming from your link `Kafka Streams Memory Management` 

That's exactly why I have a shared cache

**Comment by samclough-cs** [comment]
I'm also observing similar behaviour. Kafka Streams metrics are showing double the specified block-cache-capacity.

**Comment by nicktelford** [comment]
FYI: This was a bug in Kafka Streams that was fixed in 3.6.1: https://issues.apache.org/jira/browse/KAFKA-13973
